src/main/java/org/jenkinsci/plugins/workflow/multibranch/ResolveScmStep.java

```diff
@@ -0,0 +1,297 @@
+    @NonNull
+    public SCMSource getSource() {
+        return source;
+    }
+
+    /**
+     * Gets the {@link SCMHead} names to try and resolve.
+     *
+     * @return the {@link SCMHead} names to try and resolve.
+     */
+    @NonNull
+    public List<String> getTargets() {
+        return Collections.unmodifiableList(targets);
+    }
+
+    /**
+     * Returns {@code true} if and only if errors will be ignored.
+     *
+     * @return {@code true} if and only if errors will be ignored.
+     */
+    public boolean isIgnoreErrors() {
+        return ignoreErrors;
+    }
+
+    /**
+     * Sets the error handling behaviour.
+     *
+     * @param ignoreErrors {@code true} if and only if errors will be ignored.
+     */
+    @DataBoundSetter
+    public void setIgnoreErrors(boolean ignoreErrors) {
+        this.ignoreErrors = ignoreErrors;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public StepExecution start(StepContext context) throws Exception {
+        return new Execution(context, this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toString() {
+        return "ResolveScmStep{" +
+                "source=" + source +
+                ", targets=" + targets +
+                ", ignoreErrors=" + ignoreErrors +
+                '}';
+    }
+
+    /**
+     * Our {@link Descriptor}.
+     */
+    @Extension(optional = true)
+    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+
+        /**
+         * Default constructor.
+         */
+        public DescriptorImpl() {
+            super(Execution.class);
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public String getFunctionName() {
+            return "resolveScm";
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public String getDisplayName() {
+            return "Resolves an SCM from an SCM Source and a list of candidate target branch names";
+        }
+    }
+
+    /**
+     * Our {@link StepExecution}.
+     */
+    public static class Execution extends AbstractSynchronousStepExecution<SCM> {
+
+        /**
+         * Ensure consistent serialization.
+         */
+        private static final long serialVersionUID = 1L;
+
+        /**
+         * The step.
+         */
+        private transient final ResolveScmStep step;
+
+        /**
+         * Our constructor (avoiding Guice injection).
+         *
+         * @param context the context.
+         * @param step    the step.
+         */
+        Execution(StepContext context, ResolveScmStep step) {
+            super(context);
+            this.step = step;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        protected SCM run() throws Exception {
+            StepContext context = getContext();
+            TaskListener listener = context.get(TaskListener.class);
+            assert listener != null;
```

Comment:
If using Guice (deprecated), use `@StepContextParameter`. Otherwise you must implement `getRequiredContext`.